_includes/js/users.md

```diff
@@ -474,25 +474,49 @@ The body of the response will contain the `objectId`, `createdAt`, `sessionToken
 Linking an existing user with a service like Facebook or Twitter uses the same method \_linkWith() to associate `authData` with the user.  For example, linking a user with a Facebook account would use a request like this:
 
 ```javascript
-let myAuthData = {
+const myAuthData = {
   id: "123456789",
   "access_token": "SaMpLeAAibS7Q55FSzcERWIEmzn6rosftAr7pmDME10008bWgyZAmv7mziwfacNOhWkgxDaBf8a2a2FCc9Hbk9wAsqLYZBLR995wxBvSGNoTrEaL",
   "expiration_date": "2012-02-28T23:49:36.353Z"
 }
-let user = new Parse.User();
+const user = new Parse.User();
 user.id = "uMz0YZeAqc";
 user._linkWith("facebook", { authData: myAuthData }).then(function(user){
   // user is linked now
 });
 ```
 After linking your user to a service, you can authenticate them using matching `authData`.
 
+#### Linking un-authenticated users
+To create a link to an un-authenticated user (for example in cloud code), options can be passed to \_linkWith() to either use the `masterKey` or pass a `sessionToken`.
+
+```javascript
+const myAuthData = {
+  id: xzx5tt123,
+  access: token
+}
+
+const user = await Parse.Query(Parse.User).get(userId);
+await user._linkWith(
+  'CustomAdapter',
+  myAuthData,
+  { useMasterKey: true }
+);
+```
+
+On rest, web, mobile, or TV clients, users can then login using the `CustomAdapter` by passing `myAuthData`:
+
+```javascript
+const loggedIn = await Parse.User.logInWith('CustomAdapter', myAuthData);
```

Comment:
An example of CustomAdapter, server config and _registerProvider should be included.